sql/core/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -33,18 +33,6 @@
   private final ArrowVectorAccessor accessor;
   private ArrowColumnVector[] childColumns;
 
-  private void ensureAccessible(int index) {
-    ensureAccessible(index, 1);
-  }
-
-  private void ensureAccessible(int index, int count) {
```

Comment:
ColumnVector is a performance critical place, we don't need index checking here, like other column vector implementations.

Reply:
I agree with this in non-debug version. Can we add assert of this check at each caller site for debugging?

p.s. Sorry for slow reviews since I am on vacation this week.

Reply:
How about we do it later? We need to find a central place to put this check, instead of doing it in every implementation.

Reply:
It is good to do it later. I agree that we do the same check at one place.